solutions/devsprint-michael-douglas-2/DeliveryApp/Screens/Home/HomeView.swift

```diff
@@ -96,18 +101,32 @@ private extension HomeView {
 }
 
 extension HomeView: UITableViewDataSource {
-
+    
     public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-
+        
         return self.restaurants.count
     }
-
+    
     public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-
-        let cell = tableView.dequeueReusableCell(withIdentifier: self.restaurantCellIdentifier)!
-        cell.textLabel?.text = self.restaurants[indexPath.row].name
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: RestaurantCellView.cellIdentifier, for: indexPath) as? RestaurantCellView else {
+            return .init()
+        }
+        
+        cell.updateView(with: .init(
+            name: restaurants[indexPath.row].name,
+            detail: restaurants[indexPath.row].category,
+            icon: "restaurant-logo"))
```

Comment:
Que tal fazermos o unwrap do `restaurant` antes de utilizar?
```suggestion
        guard restaurant = restaurants[indexPath.row],
           let cell = tableView.dequeueReusableCell(withIdentifier: RestaurantCellView.cellIdentifier, for: indexPath) as? RestaurantCellView else {
            return .init()
        }
        
        cell.updateView(with: .init(
            name: restaurant.name,
            detail: restaurant.category,
            icon: "restaurant-logo")
         )
```